Add Databricks tile (#8410)
DataDog/integrations-core: databricks/README.md

# Agent Check: Databricks

## Overview

Monitor your [Databricks][1] clusters with the Datadog [Spark integration][6].

## Setup

### Installation

> Monitor Databricks Spark applications with the [Datadog Spark integration][2]. Install the Datadog Agent on your clusters following the Configuration instructions for your appropriate cluster.

Reviewer: ```suggestion
Monitor Databricks Spark applications with the [Datadog Spark integration][2]. Install the Datadog Agent on your clusters following the [Configuration](#configuration) instructions for your appropriate cluster.
```

### Configuration

Follow the Databricks-Datadog [tutorial][2] to configure the Spark integration to monitor your Apache Spark Cluster.

#### Standard cluster

Use the [Datadog Init Script][4] Databricks notebook to install the Datadog Agent and collect system and Spark metrics.

```yaml
init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$DB_DRIVER_PORT
      spark_cluster_mode: spark_standalone_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml
```

#### Job cluster

For job clusters, use the following script to configure the Spark integration.

>    **Note**: Job clusters are monitored in `spark_driver_mode` via the Spark UI port.

Reviewer: ```suggestion
**Note**: Job clusters are monitored in `spark_driver_mode` with the Spark UI port.
```

echo "Driver ip: $DB_DRIVER_IP"

cat <<EOF >> /tmp/start_datadog.sh
#!/bin/bash

if [ \$DB_IS_DRIVER ]; then
  echo "On the driver. Installing Datadog ..."

  # install the Datadog agent
  DD_API_KEY=<API_KEY> bash -c "\$(curl -L https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"

  while [ -z \$gotparams ]; do
    if [ -e "/tmp/driver-env.sh" ]; then
      DB_DRIVER_PORT=\$(grep -i "CONF_UI_PORT" /tmp/driver-env.sh | cut -d'=' -f2)
      gotparams=TRUE
    fi
    sleep 2
  done

  current=\$(hostname -I | xargs)

  # WRITING SPARK CONFIG FILE FOR STREAMING SPARK METRICS
  echo "init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$DB_DRIVER_PORT
      spark_cluster_mode: spark_driver_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml

  # RESTARTING AGENT
  sudo service datadog-agent restart

fi
EOF

# CLEANING UP
if [ \$DB_IS_DRIVER ]; then
  chmod a+x /tmp/start_datadog.sh
  /tmp/start_datadog.sh >> /tmp/datadog_start.log 2>&1 & disown
fi

```

### Validation

[Run the Agent's status subcommand][5] and look for `spark` under the Checks section.

## Data Collected

### Metrics

> See the Spark [metadata.csv][7] for a list of metrics collected.

Reviewer: ```suggestion
See the [Spark integration documentation][7] for a list of metrics collected.
```

### Service Checks

> See the Spark [service checks documentation][8] for a list of service checks collected.

Reviewer: ```suggestion
See the [Spark integration documentation][8] for the list of service checks collected.
```

 
### Events

The Databricks integration does not include any events.

## Troubleshooting

Need help? Contact [Datadog support][3].

[1]: https://databricks.com/
[2]: https://databricks.com/blog/2017/06/01/apache-spark-cluster-monitoring-with-databricks-and-datadog.html
[3]: https://docs.datadoghq.com/help/
[4]: https://docs.databricks.com/_static/notebooks/datadog-init-script.html
[5]: https://docs.datadoghq.com/agent/guide/agent-commands/?#agent-status-and-information
[6]: https://docs.datadoghq.com/integrations/spark/?tab=host

> [7]: https://github.com/DataDog/integrations-core/blob/master/spark/metadata.csv

Reviewer: ```suggestion
[7]: https://docs.datadoghq.com/integrations/spark/#metrics
```

> [8]: https://docs.datadoghq.com/integrations/spark/?tab=host#service-checks

Reviewer: ```suggestion
[8]: https://docs.datadoghq.com/integrations/spark/#service-checks
```

# Agent Check: Databricks

## Overview

This check monitors [Databricks][1].

## Setup

### Installation

> The Databricks check is included in the [Datadog Agent][2] package.

Reviewer 2: Is this the correct link? Seems to point to a blog article rather than [datadog agent link](https://docs.datadoghq.com/agent/).

Author: Yes the intention is to reference the recommended/most publicized way we currently suggest monitoring Databricks

No additional installation is needed on your server.

### Configuration

Use the [Spark Integration][2] to monitor your Apache Spark Cluster.

#### Standard Cluster

Use the [Datadog Init Script][4] Databricks notebook to install the Datadog Agent and collect system and Spark metrics.

```yaml
init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$DB_DRIVER_PORT
      spark_cluster_mode: spark_standalone_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml
```

#### Job Cluster

Modify the Spark integration configuration in the [Datadog Init Script][4] to monitor job clusters:

```yaml
init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$SPARK_UI_PORT
      spark_cluster_mode: spark_driver_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml
```

**NOTE**: The Spark UI Port is dynamically set unless you configure the `spark.ui.port` in the `Spark Config` of the Cluster Configuration page.
Be sure to create the environment variable `SPARK_UI_PORT` with the same value to use the init script.

### Validation

> [Run the Agent's status subcommand][8] and look for `spark` under the Checks section.

Reviewer 2: Reference link also missing here

# Agent Check: Databricks

## Overview

This check monitors [Databricks][1].

## Setup

### Installation

The Databricks check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

Use the [Spark Integration][2] to monitor your Apache Spark Cluster.

> #### Standard Cluster

Reviewer: ```suggestion
#### Standard cluster
```

Use the [Datadog Init Script][4] Databricks notebook to install the Datadog Agent and collect system and Spark metrics.

```yaml
init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$DB_DRIVER_PORT
      spark_cluster_mode: spark_standalone_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml
```

> #### Job Cluster

Reviewer: ```suggestion
#### Job cluster
```

Modify the Spark integration configuration in the [Datadog Init Script][4] to monitor job clusters:

```yaml
init_config:
instances:
    - spark_url: http://\$DB_DRIVER_IP:\$SPARK_UI_PORT
      spark_cluster_mode: spark_driver_mode
      cluster_name: \$current" > /etc/datadog-agent/conf.d/spark.yaml
```

> **NOTE**: The Spark UI Port is dynamically set unless you configure the `spark.ui.port` in the `Spark Config` of the Cluster Configuration page.

Reviewer: ```suggestion
**Note**: The Spark UI port is dynamically set unless you configure the `spark.ui.port` in the `Spark Config` of the cluster configuration page.
```

> Be sure to create the environment variable `SPARK_UI_PORT` with the same value to use the init script.

Reviewer: ```suggestion
Create the environment variable `SPARK_UI_PORT` with the same value to use the init script.
```

### Validation

[Run the Agent's status subcommand][5] and look for `spark` under the Checks section.

## Data Collected

### Metrics

> Databricks does not include any metrics.

Reviewer: ```suggestion
The Databricks integration does not include any metrics.
```

### Service Checks

> Databricks does not include any service checks.

Reviewer: ```suggestion
The Databricks integration does not include any service checks.
```

### Events

> Databricks does not include any events.

Reviewer: ```suggestion
The Databricks integration does not include any events.
```
